addons/website_sale/models/website_snippet_filter.py

```diff
@@ -160,34 +173,30 @@ def _get_products_recently_sold_with(self, website, limit, domain, context):
                     ).search(domain, limit=limit)
         return products
 
-    def _get_products_accessories(self, website, limit, domain, context):
-        products = []
-        current_id = context.get('product_template_id')
-        if current_id:
-            current_id = int(current_id)
-            current_template = self.env['product.template'].browse(current_id)
-            if current_template.exists():
-                excluded_products = website.sale_get_order().order_line.product_id.ids
-                excluded_products.extend(current_template.product_variant_ids.ids)
-                included_products = current_template._get_website_accessory_product().ids
-                products_ids = list(set(included_products) - set(excluded_products))
-                if products_ids:
-                    domain = expression.AND([
-                        domain,
-                        [('id', 'in', products_ids)],
-                    ])
-                    products = self.env['product.product'].with_context(
-                        display_default_code=False,
-                    ).search(domain, limit=limit)
+    def _get_products_accessories(self, website, limit, domain, product_template_id=None, **kwargs):
+        products = self.env['product.product']
+        current_template = self.env['product.template'].browse(int(product_template_id)).exists()
```

Comment:
Just a note that with the default `product_template_id=None` value, `int(None)` returns a `TypeError`, as `int` doesn't accept `None` as an argument.

Should get fixed in https://github.com/odoo/odoo/pull/179845